lib/trento/application/usecases/hosts/hosts.ex

```diff
@@ -36,4 +47,41 @@ defmodule Trento.Hosts do
         subscription_count
     end
   end
+
+  @spec deregister_host(Ecto.UUID.t(), DateService) :: :ok | {:error, any}
+  def deregister_host(host_id, date_service \\ DateService) do
+    case Repo.get_by(HostReadModel, id: host_id) do
+      nil ->
+        {:error, :host_not_found}
+
+      _ ->
+        maybe_dispatch_host_deregistration_request(host_id, date_service)
+    end
+  end
+
+  defp commanded,
+    do: Application.fetch_env!(:trento, Trento.Commanded)[:adapter]
+
+  defp maybe_dispatch_host_deregistration_request(host_id, date_service) do
```

Comment:
I think we could have a discussion about that, not on the logic, but on the place in which we should put the logic, we could add it to a a enrichable middleware, so we protect the command dispatching, making sure that the command will fail with that rule, right know we are making this rule only in the context, and the command could be dispatched without this rule.

If you take a look at others enrichable middleware, like enrich_application_instance, you can find that we are enforcing a business rule into the command dispatching.

This is more safe and aligned with the rest of codebase, it's a matter to add a new defimpl and copy the code, but I think it's more safe.

Ofcourse cc @arbulu89 @fabriziosestito

Reply:
That's a really good point @CDimonaco 
Putting in the middleware moves the logic closer to the domain, so less chances to have unwanted behaviours.
I don't think of any scenario where we don't want to enforce this logic, so I would agree that the middleware is a better place.

Reply:
@CDimonaco Very good point, I like the fact that this also protects the command from any other use outside of the usecase. We added it, but we are a bit unsure about the location we chose. It seems wrong to put it inside the `discovery` structure, but it's where the most similar code currently lives at (because other registration middleware comes from the discovery.

Do you have a better suggestion as where to place this?

Reply:
@rtorrero I think the suggestion remains the same. To implement the logic in an enrichment middleware protocol